gazebo_ros/include/gazebo_ros/node.hpp

```diff
@@ -174,6 +178,36 @@ Node::SharedPtr Node::CreateWithArgs(Args && ... args)
     static_executor_ = node->executor_;
   }
 
+  // Generate warning on start up if use_sim_time parameter is set to false
+  bool check_sim_time;
+  node->get_parameter("use_sim_time", check_sim_time);
+  if (!check_sim_time) {
+    RCLCPP_WARN(
+      node->get_logger(), "Startup warning: use_sim_time parameter will be ignored "
+      "by default plugins and ROS messages will continue to use simulation timestamps");
+  }
+
+  // Parameter change callback
+  auto param_change_callback =
+    [&node](std::vector<rclcpp::Parameter> parameters) {
```

Comment:
I'm a little worried that it's still possible for `node` to be destructed and then we try to access it asynchronously in this callback. I'm not sure if it's likely to happen (depending on how the executor handles a destructed node), but it's probably better to be safe and use a weak_ptr here instead. This way we can actually check if the node has gone out of scope or not.

Reply:
Done